From patchwork Wed Oct 10 14:46:06 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Uros Bizjak <ubizjak@gmail.com>
X-Patchwork-Id: 10634675
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 5D7665CAF
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 10 Oct 2018 14:46:37 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 383A12A547
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 10 Oct 2018 14:46:37 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 270942A54E; Wed, 10 Oct 2018 14:46:37 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id C3A8C2A547
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 10 Oct 2018 14:46:36 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726867AbeJJWJF (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 10 Oct 2018 18:09:05 -0400
Received: from mail-wm1-f66.google.com ([209.85.128.66]:53651 "EHLO
        mail-wm1-f66.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726664AbeJJWJF (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 10 Oct 2018 18:09:05 -0400
Received: by mail-wm1-f66.google.com with SMTP id y11-v6so5774188wma.3
        for <kvm@vger.kernel.org>; Wed, 10 Oct 2018 07:46:34 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id;
        bh=kau9368ggYbCL8WjoMGnHnWKfOgWTCPATHZxb61PCCU=;
        b=sDv6LkJSG0fQsPLQ4f0M4posxTgx14/NeYhYf1Rkw2acYq9UveHR+EzhUmV3UQ314j
         /r/Vn+/Ol+pukOBTNfg9dBebEQP3GoDpDrOxu/6ZvVkMEef8b/603tK9d8PfEUsHJl31
         e+hefwK948YhIlZKmRDSJZztXGE8xNHwFziUb0XhVv50LU2s4MSaSAb1AbGp83PqU5up
         oYAlsDNh3WWtUNuWrlk+m+LuLY8cpXw8P4Mvx0/PCJWZN6aGSgV9hwb0039Y5h7aM9nd
         F3oPgW2asipG2UNP8ChPs8Y3G+Wg1gSN2vdP1neQI1HEO9nrhiVBTilV5BJKXy4riAxz
         t5bA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id;
        bh=kau9368ggYbCL8WjoMGnHnWKfOgWTCPATHZxb61PCCU=;
        b=XKGc2w4TZVBWQ2tlAoGNszc3XIXeBHNd03/q6xaM3QUf+eZscOjRks+v5FWXap37EW
         wiLr980h5rm0rz0I/lTPWxNrplgGX/CpA0z5RCppq7ynsZr/pT7TuguCBgLDJ/MygEOQ
         B2/qGMjUwPwa/6DAG3oWT3n9rxf1dbiclvgprXCOEBse18xV0wJg+JJ2UCx2aaBlCsBR
         pFdByRRZci0Mu4otpeowCEGrUQT+PtLcHYNL56lsOjMxacpnC44QV0S2pe0xrj/b+IoI
         oDTie8OkhWtaYPvwXyeKXcCA4HqPqBdl3oYLSLPvTO2E+Bwtxmi+IytqZxXYe4A5L5Sl
         k1tQ==
X-Gm-Message-State: ABuFfoiQLqs6vF3CBdfIOBtzw2LtMGYsfCBX24eD9n5sNrnB3a7jurLw
        v3H9QwuwO8ipby4pAV+BkowPPP6M
X-Google-Smtp-Source: 
 ACcGV61kcsYuHMaTuS1ysU+lVwuq3FzLizjiRuWI1S5I69QSwrRDiqc68HUNsDi9QyIFOHOIWMmCSQ==
X-Received: by 2002:a1c:a90:: with SMTP id
 138-v6mr1215768wmk.49.1539182793082;
        Wed, 10 Oct 2018 07:46:33 -0700 (PDT)
Received: from localhost.localdomain.com (93-103-18-160.static.t-2.net.
 [93.103.18.160])
        by smtp.gmail.com with ESMTPSA id
 r2-v6sm3178330wrq.1.2018.10.10.07.46.32
        (version=TLS1_2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256/256);
        Wed, 10 Oct 2018 07:46:32 -0700 (PDT)
From: Uros Bizjak <ubizjak@gmail.com>
To: kvm@vger.kernel.org, pbonzini@redhat.com
Cc: Uros Bizjak <ubizjak@gmail.com>
Subject: [PATCH] KVM/x86: Fix invvpid and invept register operand size in
 64-bit mode
Date: Wed, 10 Oct 2018 16:46:06 +0200
Message-Id: <20181010144606.3645-1-ubizjak@gmail.com>
X-Mailer: git-send-email 2.17.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Register operand size of invvpid and invept instruction in 64-bit mode
has always 64 bits. Adjust inline function argument type to reflect
correct size.

Signed-off-by: Uros Bizjak <ubizjak@gmail.com>
---
 arch/x86/kvm/vmx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 612fd17be635..b04fb093ce87 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -2073,7 +2073,7 @@ static int __find_msr_index(struct vcpu_vmx *vmx, u32 msr)
 	return -1;
 }
 
-static inline void __invvpid(int ext, u16 vpid, gva_t gva)
+static inline void __invvpid(unsigned long ext, u16 vpid, gva_t gva)
 {
     struct {
 	u64 vpid : 16;
@@ -2088,7 +2088,7 @@ static inline void __invvpid(int ext, u16 vpid, gva_t gva)
     BUG_ON(error);
 }
 
-static inline void __invept(int ext, u64 eptp, gpa_t gpa)
+static inline void __invept(unsigned long ext, u64 eptp, gpa_t gpa)
 {
 	struct {
 		u64 eptp, gpa;
